jupyterlab_automl/jupyterlab_automl/service.py

```diff
@@ -266,3 +277,66 @@ def create_dataset_from_file(self, display_name, file_name, file_data):
     return self.create_dataset(display_name,
                                gcs_uri="gs://{}/{}".format(
                                    self._gcs_bucket, key))
+
+  def _get_model(self, model_id):
+    return self._model_client.get_model(name=model_id)
+  
+  def create_endpoint(self, name):
+    display_name = "ucaip-extension/" + name
+    endpoint = {"display_name": display_name}
+    return self._endpoint_client.create_endpoint(parent=self._parent, endpoint=endpoint).result()
+  
+  def delete_endpoint(self, endpoint_id):
+    self._endpoint_client.delete_endpoint(name=endpoint_id).result()
+
+  def _build_endpoint(self, deployed_model_id, endpoint):
+    return {
+        "deployedModelId": deployed_model_id,
+        "id": endpoint.name,
+        "displayName": endpoint.display_name,
+        "models": len(endpoint.deployed_models),
+        "updateTime": get_milli_time(endpoint.update_time),
+    }
+
+  def get_endpoints(self, model_id):
+    model = self._get_model(model_id)
+    endpoints = []
+    if model.deployed_models:
+      for deployed_model in model.deployed_models:
+        endpoint = self._endpoint_client.get_endpoint(name=deployed_model.endpoint)
+        built = self._build_endpoint(deployed_model.deployed_model_id, endpoint)
+        endpoints.append(built)
+    return endpoints
+
+  def check_deploying(self, model_name):
+    endpoints = self._endpoint_client.list_endpoints(parent=self._parent).endpoints
+    name = "ucaip-extension/" + model_name
+    filtered = filter(lambda x: name == x.display_name, endpoints)
+    filtered = list(filtered)
+    if len(filtered) > 0:
+      return [self._build_endpoint("None", filtered[0])]
+    return []
```

Comment:
This isn't my favorite thing but basically just checking to see if an extension managed endpoint for this model exists without a deployed model meaning that the model is currently deploying on that endpoint. I couldn't really think of a better way to do this. The alternative would be to continue to show the "deploy model" button while the model is deploying which could lead to duplicate deployments.